### ISSUE TITLE
Error upgrading from 4.0 to 4.1 with qubes-dist-upgrade

### ISSUE THREAD

**Issue by zaz** [QubesOS/qubes-issues#7838]
Running `qubes-dist-upgrade` on Qubes 4.0 fails with error `404 for https://yum.qubes-os.org/r4.0/templates-itl-testing/repodata/repomd.xml.metalink (IP: 147.75.32.1)`

```
$ sudo qubes-dist-upgrade --all -y
WARNING: /!\ ENSURE TO HAVE MADE A BACKUP OF ALL YOUR VMs AND dom0 DATA /!\
---> (STAGE 0) Skipping - already right size
---> (STAGE 1) Updating dom0...
Using sys-firewall as UpdateVM to download updates for Dom0; this may take some time...
warning: Converting database from bdb_ro to sqlite backend
47 files removed
Fedora 25 - x86_64                               11 MB/s |  50 MB     00:04
Fedora 25 - x86_64 - Updates                    4.0 MB/s |  24 MB     00:05
Qubes Dom0 Repository (updates)                 1.7 MB/s | 6.8 MB     00:03
Qubes Dom0 Repository (updates-testing)         761 kB/s | 1.5 MB     00:02
Qubes Dom0 Repository (security-testing)        178 kB/s | 1.5 MB     00:08
determining the fastest mirror (15 hosts).. done.--  B/s |   0  B     --:-- ETA
Qubes Templates repository                      2.4 kB/s | 7.1 kB     00:03
Qubes Templates repository                       40  B/s | 153  B     00:03
Errors during downloading metadata for repository 'qubes-templates-itl-testing':
  - Status code: 404 for https://yum.qubes-os.org/r4.0/templates-itl-testing/repodata/repomd.xml.metalink (IP: 147.75.32.1)
Error: Failed to download metadata for repo 'qubes-templates-itl-testing': Cannot prepare internal mirrorlist: Status code: 404 for https://yum.qubes-os.org/r4.0/templates-itl-testing/repodata/repomd.xml.metalink (IP: 147.75.32.1)
-> Launch restoration...
```

**Comment by marmarek** [comment]
That's because we're retiring R4.0 repositories: https://www.mail-archive.com/qubes-devel@googlegroups.com/msg05166.html

I've restored the metalink file, and put a redirect to the archived version. Should fix the issue without the need to modify the script.

**Comment by marmarek** [comment]
@zaz can you confirm it works now? or have you upgraded all you systems with modified script already?

**Comment by zaz** [comment]
I may have another 4.0 machine; I'll have to check.

Should the testing repo be enabled though?  It seems to me that we should not enable it unless we need it (which we don't: the script seems to work without it).

**Comment by marmarek** [comment]
Probably not necessary at this time. But also, changing this now will be rather challenging - since R4.0 is EOL already, the build server for R4.0 packages is torn down already.

**Comment by andrewdavidwong** [comment]
@marmarek, I'm inclined to say that this issue should be closed as `invalid` based on the fact that 4.0 has already reached EOL. In other words, our policy should be (and arguably already is) that we don't accept new bug reports for unsupported releases.

**Comment by andrewdavidwong** [comment]
Based on our [issue tracking guidelines](https://www.qubes-os.org/doc/issue-tracking/), this does not appear to be suitable for `qubes-issues`. We ask that you please post this on the [Qubes Forum or `qubes-users` mailing list](https://www.qubes-os.org/support/) instead. If, after reading our [issue tracking guidelines](https://www.qubes-os.org/doc/issue-tracking/), you believe we are mistaken, please leave a brief comment explaining why. We'll be happy to take another look, and, if appropriate, reopen this issue. Thank you for your understanding.
